### PR TITLE
allows schema errors to be logged

### PR DESCRIPTION
**Pull request by ryandavisflywheel**
Not having the extra handling proposed here causes another exception in RequestHandler.handle_exception when it attempts `message = str(exception)`.  My schema was not exactly correct and the actual error was not being logged due to the exception within the exception handler, which, needless to say was painful.  Hopefully this helps the next person debug more easily.

### CODE REVIEW

File: flywheel_common/providers/provider.py [hunk 1]
```diff
@@ -69,8 +69,9 @@ def validate(self):
         try:
             self._schema.validate(self._schema.dump(self).data)
         except ValidationError as e:
-            raise errors.ValidationError(str(e.messages))
-            # raise errors.ValidationError('Validation error, "{details}"')
+            # Convert to avoid format KeyErrors in ErrorBase.__str__
+            msg = str(e.messages).replace('{','{{').replace('}','}}')
```
Comment (nagem) [inline flywheel_common/providers/provider.py]: What does this message look like that requires all curly braces to be replaced by doubles?

Comment (ryandavisflywheel) [reply flywheel_common/providers/provider.py]: It has JSON schema within it so it's something akin to `{'config': {'stuff': 'thing'} }`

See for reference: https://stackoverflow.com/questions/5466451/how-can-i-print-literal-curly-brace-characters-in-python-string-and-also-use-fo

Comment (ryandavisflywheel) [reply flywheel_common/providers/provider.py]: Later on when it attempts to log debug output, 
```
class ErrorBase(Exception):
...
    def __str__(self):
        return self._msg.format(**self.__dict__)
```
blows up

Comment (nagem) [reply flywheel_common/providers/provider.py]: Ah I wasn't aware a curly brace is how you escape a curly brace, thanks.

Comment (ryandavisflywheel) [reply flywheel_common/providers/provider.py]: news to me also, I don't think I've ever needed to use that syntax ever before yesterday

**Pull request merged**